noxfile.py

```diff
@@ -366,12 +363,46 @@ def build_dists(session: nox.Session) -> List[str]:
             "Remove them and try again",
         )
 
+    session.log("# Install build environment")
+    build_venv = Path(session.create_tmp()) / "build-venv"
+    session.run("python", "-m", "venv", "--without-pip", build_venv, silent=True)
+    build_python = build_venv / "bin" / "python"
+    session.run(
+        "python",
+        "-m",
+        "pip",
+        "--python",
+        build_python,
+        "install",
+        "-r",
+        HERE / "build-requirements.txt",
+        "--no-deps",
+        "--only-binary=:all:",
+        "--require-hashes",
+        silent=True,
+    )
+
     session.log("# Build distributions")
-    session.run("python", "-m", "build", silent=True)
+    session.run(
+        build_python,
+        "-m",
+        "build",
+        "--no-isolation",
```

Comment:
Oh, is this because of `PIP_CONSTRAINT` not working with hashes and resolvelib?

Reply:
So if not for https://github.com/pypa/pip/issues/9243, https://github.com/pypa/pip/issues/4582#issuecomment-1856172641 could be used instead.